### PR TITLE
[v3] Don't bail out of instrumentation if `DD_TRACE_ENABLED=0`

### PR DESCRIPTION
**Pull request by andrewlock**
## Summary of changes

Even if `DD_TRACE_ENABLED=0`, still instrument the application, just don't send any traces etc.

## Reason for change

We are going to want to allow other products to run when APM is disabled, e.g. ASM. Currently, that's problematic because we aren't attached.

## Implementation details

Remove the bail-out clause from the native tracer. Update the AAS trace-agent logic to not check the `DD_TRACE_ENABLED` value when starting the `trace-agent`/dogstatsd (just always start them). There are very few scenarios where we _shouldn't_ start these where the native tracer is _already_ running (as it is in the managed loader)

## Test coverage

Added an integration test to confirm that we still attach, but don't send any traces when `DD_TRACE_ENABLED=0`. Fails before this PR, succeeds afterwards

## Other details

Also updated the migration docs to notify that customers should use `COR_ENABLE_PROFILING=0/CORECLR_ENABLE_PROFILING=0` if they want to bail out

One of the issues this flagged was in the profiler integration tests, where an issue related to our startup hook and the lack of debug symbols was found and fixed in https://github.com/DataDog/dd-trace-dotnet/pull/5420/commits/7ef76991198329d18c13d0240048c3fa3d82a05c

### CODE REVIEW

**Comment by andrewlock** [comment]
## Execution-Time Benchmarks Report :stopwatch:

Execution-time results for samples comparing the following branches/commits:
- [This PR (5420)](https://github.com/DataDog/dd-trace-dotnet/tree/cc271461e5fa9239e91f7fa5e40dcdbf36398b12)
- [master](https://github.com/DataDog/dd-trace-dotnet/tree/c9a65e801106916865edb87017e631b4374f2a5b)

Execution-time benchmarks measure the whole time it takes to execute a program. And are intended to measure the one-off costs. Cases where the execution time results for the PR are worse than latest master results are shown in **red**. The following thresholds were used for comparing the execution times:
* Welch test with statistical test for significance of **5%**
* Only results indicating a difference greater than **5%** and **5 ms** are considered.


Note that these results are based on a _single_ point-in-time result for each branch. For full results, see the [dashboard](https://ddstaging.datadoghq.com/dashboard/4qn-6fi-54p/apm-net-execution-time-benchmarks).

Graphs show the p99 interval based on the mean and StdDev of the test run, as well as the mean value of the run (shown as a diamond below the graph).
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET Framework 4.6.2) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (5420) - mean (72ms)  : 67, 77
     .   : milestone, 72,
    master - mean (74ms)  : 62, 85
     .   : milestone, 74,

    section CallTarget+Inlining+NGEN
    This PR (5420) - mean (1,011ms)  : 991, 1031
     .   : milestone, 1011,
    master - mean (996ms)  : 974, 1019
     .   : milestone, 996,

```
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET Core 3.1) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (5420) - mean (110ms)  : 107, 113
     .   : milestone, 110,
    master - mean (109ms)  : 106, 112
     .   : milestone, 109,

    section CallTarget+Inlining+NGEN
    This PR (5420) - mean (723ms)  : 705, 742
     .   : milestone, 723,
    master - mean (713ms)  : 684, 742
     .   : milestone, 713,

```
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET 6) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (5420) - mean (94ms)  : 90, 97
     .   : milestone, 94,
    master - mean (93ms)  : 90, 97
     .   : milestone, 93,

    section CallTarget+Inlining+NGEN
    This PR (5420) - mean (688ms)  : 659, 716
     .   : milestone, 688,
    master - mean (670ms)  : 647, 693
     .   : milestone, 670,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET Framework 4.6.2) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (5420) - mean (187ms)  : 184, 191
     .   : milestone, 187,
    master - mean (187ms)  : 183, 191
     .   : milestone, 187,

    section CallTarget+Inlining+NGEN
    This PR (5420) - mean (1,087ms)  : 1062, 1113
     .   : milestone, 1087,
    master - mean (1,068ms)  : 1044, 1092
     .   : milestone, 1068,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET Core 3.1) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (5420) - mean (271ms)  : 268, 275
     .   : milestone, 271,
    master - mean (270ms)  : 263, 276
     .   : milestone, 270,

    section CallTarget+Inlining+NGEN
    This PR (5420) - mean (880ms)  : 856, 905
     .   : milestone, 880,
    master - mean (870ms)  : 845, 895
     .   : milestone, 870,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET 6) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (5420) - mean (261ms)  : 257, 264
     .   : milestone, 261,
    master - mean (259ms)  : 254, 264
     .   : milestone, 259,

    section CallTarget+Inlining+NGEN
    This PR (5420) - mean (870ms)  : 849, 891
     .   : milestone, 870,
    master - mean (849ms)  : 827, 872
     .   : milestone, 849,

```

File: tracer/src/Datadog.Trace/AgentProcessManager.cs [hunk 1]
```diff
@@ -74,39 +74,30 @@ public static void Initialize()
...
-                    {
-                        Log.Warning("Directory for trace agent does not exist: {Directory}. The process won't be started.", TraceAgentMetadata.DirectoryPath);
-                    }
-                    else
-                    {
-                        Processes.Add(TraceAgentMetadata);
-                    }
+                    Log.Warning("Requested to start the Trace Agent but the process path hasn't been supplied in environment.");
+                }
+                else if (!Directory.Exists(TraceAgentMetadata.DirectoryPath))
```
Comment (lucaspimentel) [inline tracer/src/Datadog.Trace/AgentProcessManager.cs:81]: This is not part of your changes, but shouldn't we also check if the executable file exists, not only the directory?

File: tracer/src/Datadog.Tracer.Native/cor_profiler.cpp [hunk 2]
```diff
@@ -52,21 +52,8 @@ HRESULT STDMETHODCALLTYPE CorProfiler::Initialize(IUnknown* cor_profiler_info_un
...
-                "DATADOG TRACER DIAGNOSTICS - In AAS, automatic tracing is disabled but keeping the profiler up to start child processes.");
-        }
-        else
-        {
-            Logger::Info("DATADOG TRACER DIAGNOSTICS - Profiler disabled in ", environment::tracing_enabled);
-            return CORPROF_E_PROFILER_CANCEL_ACTIVATION;
-        }
-    }
+    // we used to bail-out if tracing was disabled, but we now allow the tracer to be loaded
+    // in all cases, so that we can enabled other products
```
Comment (lucaspimentel) [inline tracer/src/Datadog.Tracer.Native/cor_profiler.cpp:56]:
```suggestion
    // in all cases, so that we can enable other products
```

**Review by lucaspimentel — APPROVED** [review]

**Comment by andrewlock** [comment]
## Benchmarks Report for appsec :snail:

Benchmarks for #5420 compared to [master](https://github.com/DataDog/dd-trace-dotnet/tree/f402602adf0ec05d83541ecd1eac42b3e4b39b75): 
* 1 benchmarks are **faster**, with geometric mean 1.846
* 1 benchmarks are **slower**, with geometric mean 1.250
* 1 benchmarks have **fewer** allocations
* 1 benchmarks have **more** allocations


The following thresholds were used for comparing the benchmark speeds:
* Mann–Whitney U test with statistical test for significance of **5%**
* Only results indicating a difference greater than **10%** and **0.3 ns** are considered.

Allocation changes below **0.5%** are ignored.


### Benchmark details


<details><summary><strong>Benchmarks.Trace.Asm.AppSecBodyBenchmark</strong> - Slower :warning: Same allocations :heavy_check_mark:



#### Slower :warning: in #5420

|  Benchmark         | diff/base | Base Median (ns) | Diff Median (ns) | Modality   |
|:----------|-----------:|-----------:|--------:|--------:|
| Benchmarks.Trace.Asm.AppSecBodyBenchmark.ObjectExtractorSimpleBody&#8209;netcoreapp3.1 | 1.250 | 204.49 | 255.68 | 

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f402602adf0ec05d83541ecd1eac42b3e4b39b75)|`AllCycleSimpleBody`|net6.0|72.7μs|199ns|772ns|0.0728| 0|0|5.99&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f402602adf0ec05d83541ecd1eac42b3e4b39b75)|`AllCycleSimpleBody`|netcoreapp3.1|61.8μs|57.8ns|216ns|0.0934| 0|0|6.93&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f402602adf0ec05d83541ecd1eac42b3e4b39b75)|`AllCycleSimpleBody`|net472|49.1μs|60.2ns|233ns|1.32| 0|0|8.32&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f402602adf0ec05d83541ecd1eac42b3e4b39b75)|`AllCycleMoreComplexBody`|net6.0|77.4μs|100ns|388ns|0.117| 0|0|9.5&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f402602adf0ec05d83541ecd1eac42b3e4b39b75)|`AllCycleMoreComplexBody`|netcoreapp3.1|68.6μs|77.7ns|301ns|0.103| 0|0|10.35&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f402602adf0ec05d83541ecd1eac42b3e4b39b75)|`AllCycleMoreComplexBody`|net472|54.8μs|48.2ns|187ns|1.86| 0.0274|0|11.83&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f402602adf0ec05d83541ecd1eac42b3e4b39b75)|`ObjectExtractorSimpleBody`|net6.0|154ns|0.0972ns|0.376ns|0.0039| 0|0|280&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f402602adf0ec05d83541ecd1eac42b3e4b39b75)|`ObjectExtractorSimpleBody`|netcoreapp3.1|205ns|0.167ns|0.647ns|0.00371| 0|0|272&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f402602adf0ec05d83541ecd1eac42b3e4b39b75)|`ObjectExtractorSimpleBody`|net472|159ns|0.113ns|0.437ns|0.0446| 0|0|281&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f402602adf0ec05d83541ecd1eac42b3e4b39b75)|`ObjectExtractorMoreComplexBody`|net6.0|2.98μs|5.11ns|19.8ns|0.0526| 0|0|3.78&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f402602adf0ec05d83541ecd1eac42b3e4b39b75)|`ObjectExtractorMoreComplexBody`|netcoreapp3.1|4.09μs|2.53ns|9.81ns|0.049| 0|0|3.69&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f402602adf0ec05d83541ecd1eac42b3e4b39b75)|`ObjectExtractorMoreComplexBody`|net472|4.39μs|3.45ns|13.4ns|0.602| 0.00656|0|3.8&nbsp;KB|
|#5420|`AllCycleSimpleBody`|net6.0|71.6μs|73.9ns|286ns|0.0714| 0|0|5.99&nbsp;KB|
|#5420|`AllCycleSimpleBody`|netcoreapp3.1|62.2μs|94.6ns|366ns|0.0622| 0|0|6.93&nbsp;KB|
|#5420|`AllCycleSimpleBody`|net472|48.2μs|55.2ns|206ns|1.32| 0|0|8.32&nbsp;KB|
|#5420|`AllCycleMoreComplexBody`|net6.0|77.9μs|84.2ns|315ns|0.117| 0|0|9.49&nbsp;KB|
|#5420|`AllCycleMoreComplexBody`|netcoreapp3.1|69μs|85.9ns|333ns|0.138| 0|0|10.35&nbsp;KB|
|#5420|`AllCycleMoreComplexBody`|net472|55μs|51.2ns|198ns|1.86| 0.0277|0|11.83&nbsp;KB|
|#5420|`ObjectExtractorSimpleBody`|net6.0|145ns|0.1ns|0.374ns|0.00391| 0|0|280&nbsp;B|
|#5420|`ObjectExtractorSimpleBody`|netcoreapp3.1|256ns|0.191ns|0.741ns|0.00374| 0|0|272&nbsp;B|
|#5420|`ObjectExtractorSimpleBody`|net472|160ns|0.11ns|0.425ns|0.0446| 0|0|281&nbsp;B|
|#5420|`ObjectExtractorMoreComplexBody`|net6.0|3.02μs|0.991ns|3.71ns|0.0528| 0|0|3.78&nbsp;KB|
|#5420|`ObjectExtractorMoreComplexBody`|netcoreapp3.1|4.14μs|2.17ns|8.4ns|0.0499| 0|0|3.69&nbsp;KB|
|#5420|`ObjectExtractorMoreComplexBody`|net472|4.13μs|1.43ns|5.56ns|0.602| 0.00616|0|3.8&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.Asm.AppSecEncoderBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f402602adf0ec05d83541ecd1eac42b3e4b39b75)|`EncodeArgs`|net6.0|39.7μs|20ns|77.6ns|0.454| 0|0|32.4&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f402602adf0ec05d83541ecd1eac42b3e4b39b75)|`EncodeArgs`|netcoreapp3.1|55.5μs|21.3ns|82.5ns|0.443| 0|0|32.4&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f402602adf0ec05d83541ecd1eac42b3e4b39b75)|`EncodeArgs`|net472|70.4μs|53ns|205ns|5.16| 0.0701|0|32.5&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f402602adf0ec05d83541ecd1eac42b3e4b39b75)|`EncodeLegacyArgs`|net6.0|75.8μs|26.1ns|97.5ns|0| 0|0|2.14&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f402602adf0ec05d83541ecd1eac42b3e4b39b75)|`EncodeLegacyArgs`|netcoreapp3.1|106μs|505ns|2.02μs|0| 0|0|2.14&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f402602adf0ec05d83541ecd1eac42b3e4b39b75)|`EncodeLegacyArgs`|net472|157μs|89.4ns|334ns|0.313| 0|0|2.15&nbsp;KB|
|#5420|`EncodeArgs`|net6.0|39.9μs|19.2ns|71.7ns|0.463| 0|0|32.4&nbsp;KB|
|#5420|`EncodeArgs`|netcoreapp3.1|55.5μs|28.5ns|111ns|0.417| 0|0|32.4&nbsp;KB|
|#5420|`EncodeArgs`|net472|70.1μs|23.9ns|92.7ns|5.16| 0.0698|0|32.5&nbsp;KB|
|#5420|`EncodeLegacyArgs`|net6.0|69.3μs|19.6ns|73.2ns|0.0346| 0|0|2.14&nbsp;KB|
|#5420|`EncodeLegacyArgs`|netcoreapp3.1|104μs|154ns|595ns|0| 0|0|2.15&nbsp;KB|
|#5420|`EncodeLegacyArgs`|net472|155μs|85.1ns|318ns|0.311| 0|0|2.15&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.Asm.AppSecWafBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f402602adf0ec05d83541ecd1eac42b3e4b39b75)|`RunWafRealisticBenchmark`|net6.0|183μs|75.8ns|294ns|0| 0|0|2.13&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f402602adf0ec05d83541ecd1eac42b3e4b39b75)|`RunWafRealisticBenchmark`|netcoreapp3.1|197μs|196ns|733ns|0| 0|0|2.11&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f402602adf0ec05d83541ecd1eac42b3e4b39b75)|`RunWafRealisticBenchmark`|net472|215μs|120ns|465ns|0.323| 0|0|2.17&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f402602adf0ec05d83541ecd1eac42b3e4b39b75)|`RunWafRealisticBenchmarkWithAttack`|net6.0|121μs|75.9ns|284ns|0| 0|0|1.37&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f402602adf0ec05d83541ecd1eac42b3e4b39b75)|`RunWafRealisticBenchmarkWithAttack`|netcoreapp3.1|129μs|139ns|540ns|0| 0|0|1.36&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f402602adf0ec05d83541ecd1eac42b3e4b39b75)|`RunWafRealisticBenchmarkWithAttack`|net472|143μs|88.5ns|343ns|0.213| 0|0|1.39&nbsp;KB|
|#5420|`RunWafRealisticBenchmark`|net6.0|184μs|156ns|603ns|0| 0|0|2.13&nbsp;KB|
|#5420|`RunWafRealisticBenchmark`|netcoreapp3.1|197μs|36.9ns|143ns|0| 0|0|2.1&nbsp;KB|
|#5420|`RunWafRealisticBenchmark`|net472|215μs|51.6ns|200ns|0.324| 0|0|2.17&nbsp;KB|
|#5420|`RunWafRealisticBenchmarkWithAttack`|net6.0|121μs|42ns|163ns|0| 0|0|1.37&nbsp;KB|
|#5420|`RunWafRealisticBenchmarkWithAttack`|netcoreapp3.1|130μs|93.5ns|362ns|0| 0|0|1.36&nbsp;KB|
|#5420|`RunWafRealisticBenchmarkWithAttack`|net472|142μs|38.5ns|149ns|0.212| 0|0|1.39&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.Iast.StringAspectsBenchmark</strong> - Faster :tada: More allocations :warning:



#### Faster :tada: in #5420

|  Benchmark         | base/diff | Base Median (ns) | Diff Median (ns) | Modality   |
|:----------|-----------:|-----------:|--------:|--------:|
| Benchmarks.Trace.Iast.StringAspectsBenchmark.StringConcatAspectBenchmark&#8209;net6.0 | 1.846 | 533,050.00 | 288,800.00 | 



#### More allocations :warning: in #5420

|  Benchmark         | Base Allocated | Diff Allocated | Change   | Change % |
|:----------|-----------:|-----------:|--------:|--------:|
| Benchmarks.Trace.Iast.StringAspectsBenchmark.StringConcatAspectBenchmark&#8209;netcoreapp3.1 | 201.74 KB | 215.24 KB | 13.5 KB | 6.69%



#### Fewer allocations :tada: in #5420

|  Benchmark         | Base Allocated | Diff Allocated | Change   | Change % |
|:----------|-----------:|-----------:|--------:|--------:|
| Benchmarks.Trace.Iast.StringAspectsBenchmark.StringConcatAspectBenchmark&#8209;net6.0 | 265.85 KB | 203.42 KB | -62.43 KB | -23.48%

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f402602adf0ec05d83541ecd1eac42b3e4b39b75)|`StringConcatBenchmark`|net6.0|60.1μs|487ns|4.46μs|0| 0|0|43.44&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f402602adf0ec05d83541ecd1eac42b3e4b39b75)|`StringConcatBenchmark`|netcoreapp3.1|54.2μs|187ns|814ns|0| 0|0|42.64&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f402602adf0ec05d83541ecd1eac42b3e4b39b75)|`StringConcatBenchmark`|net472|38.1μs|126ns|454ns|0| 0|0|59.07&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f402602adf0ec05d83541ecd1eac42b3e4b39b75)|`StringConcatAspectBenchmark`|net6.0|540μs|2.96μs|17.8μs|0| 0|0|265.85&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f402602adf0ec05d83541ecd1eac42b3e4b39b75)|`StringConcatAspectBenchmark`|netcoreapp3.1|293μs|2.35μs|23.1μs|0| 0|0|201.74&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/f402602adf0ec05d83541ecd1eac42b3e4b39b75)|`StringConcatAspectBenchmark`|net472|256μs|4.69μs|45μs|0| 0|0|221.18&nbsp;KB|
|#5420|`StringConcatBenchmark`|net6.0|59.1μs|749ns|7.3μs|0| 0|0|43.44&nbsp;KB|
|#5420|`StringConcatBenchmark`|netcoreapp3.1|54μs|255ns|955ns|0| 0|0|42.64&nbsp;KB|
|#5420|`StringConcatBenchmark`|net472|37.9μs|127ns|474ns|0| 0|0|59.07&nbsp;KB|
|#5420|`StringConcatAspectBenchmark`|net6.0|288μs|1.22μs|4.41μs|0| 0|0|203.42&nbsp;KB|
|#5420|`StringConcatAspectBenchmark`|netcoreapp3.1|325μs|1.78μs|10.2μs|0| 0|0|215.24&nbsp;KB|
|#5420|`StringConcatAspectBenchmark`|net472|266μs|5.29μs|50.2μs|0| 0|0|221.18&nbsp;KB|
</details>

**Comment by andrewlock** [comment]
## Throughput/Crank Report :zap:

Throughput results for AspNetCoreSimpleController comparing the following branches/commits:
- [This PR (5420)](https://github.com/DataDog/dd-trace-dotnet/tree/0e9a5a3e918de22a20e7b6e0a4ab4352fec41311)
- [master](https://github.com/DataDog/dd-trace-dotnet/tree/5684bf9389276d92f0862c1ebdab3afdfc343296)
- [benchmarks/2.9.0](https://github.com/DataDog/dd-trace-dotnet/tree/89ae3f4ee7d85b824badda5873166707c9e560b4)

Cases where throughput results for the PR are worse than latest master (5% drop or greater), results are shown in **red**.

Note that these results are based on a _single_ point-in-time result for each branch. For full results, see [one](https://ddstaging.datadoghq.com/dashboard/fnz-afi-c2i/apm-net-crank) of the [many](https://ddstaging.datadoghq.com/dashboard/c5g-i7d-kcy/ciapp-apm-net-throughput-tests), [many](https://ddstaging.datadoghq.com/dashboard/uxh-m8j-qhi/ciapp-apm-net-throughput-tests-kevin) dashboards!

```mermaid
gantt
    title Throughput Linux x64 (Total requests) 
    dateFormat  X
    axisFormat %s
    section Baseline
    This PR (5420) (10.778M)   : 0, 10778003
    master (11.174M)   : 0, 11174144
    benchmarks/2.9.0 (11.327M)   : 0, 11326637

    section Automatic
    This PR (5420) (7.480M)   : 0, 7479771
    master (7.680M)   : 0, 7680067
    benchmarks/2.9.0 (8.199M)   : 0, 8199353

    section Trace stats
    This PR (5420) (7.853M)   : 0, 7853449
    master (7.968M)   : 0, 7968261

    section Manual
    This PR (5420) (10.966M)   : 0, 10965512
    master (9.672M)   : 0, 9672287

    section Manual + Automatic
    This PR (5420) (7.171M)   : 0, 7171367
    master (7.144M)   : 0, 7143638

```
```mermaid
gantt
    title Throughput Linux arm64 (Total requests) 
    dateFormat  X
    axisFormat %s
    section Baseline
    This PR (5420) (9.675M)   : 0, 9674891
    master (9.450M)   : 0, 9450329
    benchmarks/2.9.0 (9.520M)   : 0, 9520488

    section Automatic
    This PR (5420) (6.567M)   : 0, 6566726
    master (6.717M)   : 0, 6716595

    section Trace stats
    This PR (5420) (6.944M)   : 0, 6943640
    master (7.018M)   : 0, 7018322

    section Manual
    This PR (5420) (9.441M)   : 0, 9440910
    master (8.273M)   : 0, 8272823

    section Manual + Automatic
    This PR (5420) (6.037M)   : 0, 6036878
    master (6.223M)   : 0, 6222897

```
```mermaid
gantt
    title Throughput Windows x64 (Total requests) 
    dateFormat  X
    axisFormat %s
    section Baseline
    This PR (5420) (10.382M)   : 0, 10382207
    master (9.945M)   : 0, 9944630
    benchmarks/2.9.0 (9.798M)   : 0, 9797854

    section Automatic
    This PR (5420) (7.169M)   : 0, 7169400
    master (7.216M)   : 0, 7215954
    benchmarks/2.9.0 (7.323M)   : 0, 7323469

    section Trace stats
    This PR (5420) (7.529M)   : 0, 7529116
    master (7.495M)   : 0, 7495479

    section Manual
    This PR (5420) (9.966M)   : 0, 9965647
    master (9.061M)   : 0, 9060687

    section Manual + Automatic
    This PR (5420) (6.849M)   : 0, 6848873
    master (6.003M)   : 0, 6002934

```

File: tracer/src/Datadog.Trace.ClrProfiler.Managed.Loader/Startup.cs [hunk 3]
```diff
@@ -62,30 +58,18 @@ static Startup()
...
+            if (runInAas)
             {
+                // With V3, pretty much all scenarios require the trace-agent and dogstatsd, so we enable them by default
                 StartupLogger.Log("Invoking managed method to start external processes.");
                 TryInvokeManagedMethod("Datadog.Trace.AgentProcessManager", "Initialize", "Datadog.Trace.AgentProcessManagerLoader");
-            }
 
-            if (automaticTraceEnabled)
-            {
                 StartupLogger.Log("Invoking managed tracer.");
```
Comment (bouwkast) [inline tracer/src/Datadog.Trace.ClrProfiler.Managed.Loader/Startup.cs:67]: I think this one needs to go down outside of this `if`

**Review by bouwkast — APPROVED** [review]
Looks good to me

**Review by daniel-romano-DD — APPROVED** [review]
LGTM. TY

File: tracer/src/Datadog.Trace/AgentProcessManager.cs (hunk 1 above)
Comment (andrewlock) [reply tracer/src/Datadog.Trace/AgentProcessManager.cs:81]: Yep, makes sense, I'll add check for that 👍

Comment (andrewlock) [reply tracer/src/Datadog.Trace/AgentProcessManager.cs:81]: Done in `StartProcesses()`

**Pull request merged**